From f5c5d0794bff5dc9034e46b3c0d273562892f58f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 3 Jul 2020 14:48:18 +0200
Subject: [PATCH 46/47] ucm: fix parse_get_safe_name() - safe name must be
 checked after substitution

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
(cherry picked from commit 30d12e930cfcba777a29af502a9c1dbfa5e37c04)
Signed-off-by: Hui Wang <hui.wang@canonical.com>
---
 src/ucm/parser.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/ucm/parser.c b/src/ucm/parser.c
index a177fd2b..570eec5e 100644
--- a/src/ucm/parser.c
+++ b/src/ucm/parser.c
@@ -234,9 +234,14 @@ int parse_get_safe_name(snd_use_case_mgr_t *uc_mgr, snd_config_t *n,
 		if (err < 0)
 			return err;
 	}
-	if (!parse_is_name_safe(id))
+	err = get_string3(uc_mgr, id, name);
+	if (err < 0)
+		return err;
+	if (!parse_is_name_safe(*name)) {
+		free(*name);
 		return -EINVAL;
-	return get_string3(uc_mgr, id, name);
+	}
+	return 0;
 }
 
 /*
-- 
2.25.1

